common/dbconfig.h

```diff
@@ -0,0 +1,60 @@
+// Copyright 2021-present Pinterest. All Rights Reserved.
+//
+// @author Premkumar (prem@pinterest.com)
+//
+
+#pragma once
+
+#include <memory>
+#include <string>
+#include <unordered_map>
+
+#include "common/config.h"
+#include "common/jsoncpp/include/json/json.h"
+
+/**
+ * {
+ *   "dataset" : {
+ *      // dataset level configs
+ *        "ad_query_join" : {
+ *              "ack_mode" : 1,
+ *       }
+ *    }
+ * }
+ */
+
+namespace common {
+
+struct DBConfig {
+  std::unordered_map<std::string, Config> dataSetConfigMap;
+};
+
+class DBConfigManager {
+public:
+  static DBConfigManager* get();
+
+
+  /**
+   * Both the load functions are used to load data after instantiation
+   * During init there may not be data for it to load.
+   **/
+  bool hasValidData() const {
+    return fDataLoaded;
+  };
+  /* load from a json object */
+  bool loadJsonObject(const Json::Value& root);
+  /* load from json string */
+  bool loadJsonStr(const std::string& data);
+
+  /* conf getter functions */
+  uint getReplicationMode(const std::string& dbName, uint defValue = 0) const;
+
+private:
+  /* No direct instantiation. Use the global static via ::get()*/
+  DBConfigManager();
+  std::shared_ptr<const DBConfig> dbConfig;
+  const Config* getConfig(const std::string& dbName) const;
+  std::shared_ptr<const DBConfig> parseConfig(const Json::Value& root);
+  bool fDataLoaded = false;
```

Comment:
can you append "_" to all data members to follow the naming convention?

Reply:
Done